editor/editor_node.cpp

```diff
@@ -7591,36 +7590,52 @@ EditorNode::EditorNode() {
 		title_bar->add_child(right_menu_spacer);
 	}
 
-	String current_renderer = GLOBAL_GET("rendering/renderer/rendering_method");
+	String current_renderer_ps = GLOBAL_GET("rendering/renderer/rendering_method");
+	current_renderer_ps = current_renderer_ps.to_lower();
+	String current_renderer_os = OS::get_singleton()->get_current_rendering_method().to_lower();
 
-	PackedStringArray renderers = ProjectSettings::get_singleton()->get_custom_property_info().get(StringName("rendering/renderer/rendering_method")).hint_string.split(",", false);
+	if (current_renderer_ps == current_renderer_os) {
+		renderer->connect("item_selected", callable_mp(this, &EditorNode::_renderer_selected));
+		// As we are doing string comparisons, keep in standard case to prevent problems with capitals
+		// "vulkan" in particular uses lowercase "v" in the code, and uppercase in the UI.
+		PackedStringArray renderers = ProjectSettings::get_singleton()->get_custom_property_info().get(StringName("rendering/renderer/rendering_method")).hint_string.split(",", false);
+		for (int i = 0; i < renderers.size(); i++) {
+			String rendering_method = renderers[i];
 
-	// As we are doing string comparisons, keep in standard case to prevent problems with capitals
-	// "vulkan" in particular uses lowercase "v" in the code, and uppercase in the UI.
-	current_renderer = current_renderer.to_lower();
+			// Add the renderers name to the UI.
+			if (rendering_method == "forward_plus") {
+				renderer->add_item(TTR("Forward+"));
+			}
+			if (rendering_method == "mobile") {
+				renderer->add_item(TTR("Mobile"));
+			}
+			if (rendering_method == "gl_compatibility") {
+				renderer->add_item(TTR("Compatibility"));
+			}
+			renderer->set_item_metadata(i, rendering_method);
 
-	for (int i = 0; i < renderers.size(); i++) {
-		String rendering_method = renderers[i];
+			// Lowercase for standard comparison.
+			rendering_method = rendering_method.to_lower();
 
-		// Add the renderers name to the UI.
-		if (rendering_method == "forward_plus") {
-			renderer->add_item(TTR("Forward+"));
+			if (current_renderer_ps == rendering_method) {
+				renderer->select(i);
+				renderer_current = i;
+			}
 		}
-		if (rendering_method == "mobile") {
-			renderer->add_item(TTR("Mobile"));
+	} else {
+		// It's an CLI-overridden rendering method.
+		if (current_renderer_os == "forward_plus") {
+			renderer->add_item(TTR("Forward+ overridden"));
 		}
-		if (rendering_method == "gl_compatibility") {
-			renderer->add_item(TTR("Compatibility"));
+		if (current_renderer_os == "mobile") {
+			renderer->add_item(TTR("Mobile overridden"));
 		}
-		renderer->set_item_metadata(i, rendering_method);
-
-		// Lowercase for standard comparison.
-		rendering_method = rendering_method.to_lower();
-
-		if (current_renderer == rendering_method) {
-			renderer->select(i);
-			renderer_current = i;
+		if (current_renderer_os == "gl_compatibility") {
+			renderer->add_item(TTR("Compatibility overridden"));
 		}
```

Comment:
Above the `if (current_renderer_ps == current_renderer_os) {` you could add this (not checked for successful compile; feel free to fix) and use it in both branches of the `if (current_renderer_ps == current_renderer_os) {`, with `p_overridden` being `false`/`true` accordingly:
```C++
auto _add_renderer_entry = [renderer](const String& p_renderer_name, bool p_mark_overridden) {
		String item_text;
		if (p_renderer_name == "forward_plus") {
			item_text = TTR("Forward+");
		} else if (p_renderer_name == "mobile") {
			item_text = TTR("Mobile");
		} else if (p_renderer_name == "gl_compatibility") {
			item_text = TTR("Compatibility");
		} else {
			item_text = p_renderer_name; // Fallback to add unbeautified.
		}
		if (p_overridden) {
			item_text += " " + TTR("Overridden");
		}
		renderer->add_item(item_text);
};
```

Please ask me for any clarification.

Reply:
I like the lambda approach, but I though godot forbids lambda?

Reply:
They are allowed if the situation well justifies it. I've used them in cases like this, where they avoid code duplication, with the extra benefits of avoiding having to add a new member function, which also helps readability, as the lambda is in the scope where it's used.